spec/draft/profile-spec.md

```diff
@@ -340,6 +340,85 @@ model User {
 }
 ```
 
+# Example
+
+Example : example Identifier { ExampleInput ExampleOutput }
+
+ExampleInput : input ComlinkLiteral
+
+ExampleOutput : 
+- ExampleResult
+- ExampleError
+
+ExampleResult : result ComlinkLiteral
+
+ExampleError : error ComlinkLiteral
+
+# Literal
+
+## Comlink Literal
+
+ComlinkLiteral :
+- PrimitiveLiteral
+- ObjectLiteral
+- ArrayLiteral
+
+## Primitive Literal
+
+PrimitiveLiteral : StringValue | NumberLiteral | BooleanLiteral | NullLiteral
+
+NumberLiteral: NumberSign? NumberLiteralDigits
+
+NumberSign : `-` | `+`
+
+NumberLiteralDigits : NumberLiteralInteger | NumberLiteralFloat
+
+NumberLiteralInteger : NumberLiteralIntegerBaseTen | NumberLiteralIntegerBaseTwo | NumberLiteralIntegerBaseEight | NumberLiteralIntegerBaseSixteen
+
+NumberLiteralIntegerBaseTen : DigitBaseTen+
+
+NumberLiteralIntegerBaseTwo : 0b DigitBaseTwo+
+
+NumberLiteralIntegerBaseEight : 0o DigitBaseEight+
+
+NumberLiteralIntegerBaseSixteen : 0x DigitBaseSixteen+
+
+DigitBaseTen : /[0-9]/
+
+DigitBaseTwo : /[0-1]/
+
+DigitBaseEight : /[0-7]/
+
+DigitBaseSixteen : /[0-9a-fA-F]/
+
+NumberLiteralFloat : DigitBaseTen+ . DigitBaseTen+
+
+BooleanLiteral : `true` | `false`
+
+NullLiteral : `null`
```

Comment:
So we decided to go with the null literal? This will need to be implemented in both AST and parser.

Currently the AST supports leaving out `input` and `result` sections alltogether, so that together with not specifying keys in objects literals could server as absence-of-value indicator.

A possible problem is with arrays, where we might want to specify a null/undefined element explicitly, so maybe we'll need an explicit null/undefined.

Reply:
If it's not implemented at this point, why don't we leave it out and put it in a future draft when we're ready to implement it.